### PR TITLE
Update fragments example to use JSX syntax

### LINKED ISSUE #518: Update react-component.html for React 16.2's improved fragment support
Hi!

React 16.2.0 [improved support for fragments](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html), and #392 added some documentation in this repo for them.

However I've spotted another reference to fragments that is still using the array syntax - that I think should probably be updated too?
https://reactjs.org/docs/react-component.html#fragments

### PR DESCRIPTION
**Pull request by gnapse**
Closes #518

### CODE REVIEW

**Comment by gaearon** [comment]
How do you feel about showing both? We should still document that arrays "work" even if they're not the most ergonomic way to do it.

**Comment by gnapse** [comment]
Definitely, was thinking about that too after I replaced it.

File: content/docs/reference-react-component.md [hunk 1]
```diff
@@ -122,13 +122,27 @@ You can also return multiple items from `render()` using an array:
...
-    <li key="B">Second item</li>,
-    <li key="C">Third item</li>,
+    <li key="A">First item</li>,
+    <li key="B">Second item</li>,
+    <li key="C">Third item</li>,
   ];
 }
 ```
 
+The same can also be accomplished using [fragments](/docs/fragments.html):
```
Comment (gaearon) [inline content/docs/reference-react-component.md]: ... which don't require keys for static items:

(and then let's remove keys)

Comment (alexkrolick) [reply content/docs/reference-react-component.md]: Should there be a note here "New in React 16.2"?

Comment (gnapse) [reply content/docs/reference-react-component.md]: The "New in React 16.2" I assumed was already covered in the linked document about details of this feature. Didn't want to attempt to cover too much detail here.

Comment (gaearon) [reply content/docs/reference-react-component.md]: We can say "Since React 16.2.0, ..."

"React 16.2.0" could be a link to the blog post

**Comment by gaearon** [comment]
Thanks!

**Comment by gnapse** [comment]
Thank you for the opportunity and the guidance!

**Pull request merged**